src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs

```diff
@@ -93,17 +95,35 @@ private bool IsLikelyEntryPointName(string name, Document document)
             // if it has the 'Async' suffix, and remove that suffix if so.
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var methodSymbolOpt = semanticModel.GetDeclaredSymbol(node) as IMethodSymbol;
+            var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            var knownTypes = new KnownTypes(compilation);
 
             bool isOrdinaryOrLocalFunction = methodSymbolOpt.IsOrdinaryMethodOrLocalFunction();
-            if (isOrdinaryOrLocalFunction && !methodSymbolOpt.Name.EndsWith(AsyncSuffix) && !isEntryPoint)
+            if (isOrdinaryOrLocalFunction
+                && !methodSymbolOpt.Name.EndsWith(AsyncSuffix)
+                && !isEntryPoint
+                && NeedsRename(this, methodSymbolOpt, keepVoid, in knownTypes))
```

Comment:
alternatively, move the other checks into NeedsRename.  Then the 'if' statement is a bit clearer about whats going on.

Reply:
Updated